src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,55 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
-
+        final String operationType1 = "supply";
+        final String operationType2 = "buy";
+        final String operationType3 = "result";
+        File importedFile = new File(fromFileName);
+        File exportedFile = new File(toFileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(importedFile))) {
+            String data = bufferedReader.readLine();
+            while (data != null) {
+                stringBuilder.append(data).append(System.lineSeparator());
+                data = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to read from this file", e);
+        }
+        String[] filteredData = stringBuilder.toString().split(System.lineSeparator());
+        String[] data = new String[filteredData.length * 2];
```

Comment:
The creation of the `data` array is unnecessary and inefficient. You can directly process `filteredData` without creating a new array, as it already contains the necessary information split by lines.